tests/AggregationBlogPost/article.md

```diff
@@ -0,0 +1,713 @@
+        count: Expression::intFieldPath('count'),
+        lowest: Expression::fieldPath('lowest'),
+        highest: Expression::fieldPath('highest'),
+        average: Expression::fieldPath('average'),
+    ),
+    self::groupBrandsAndSort(
+        _id: object(
+            year: Expression::fieldPath('_id.year'),
+            month: Expression::fieldPath('_id.month'),
+        ),
+        brand: Expression::fieldPath('_id.brand'),
+        count: Expression::fieldPath('count'),
+        prices: Expression::fieldPath('prices'),
+    ),
+);
+```
+
+### Extract Complex Expressions
+
+So far, we've only extracted entire pipeline stages that contain relatively
+simple expressions. Sometimes your aggregation pipeline will contain a more
+complex expression. From the same project that I took the previous example from,
+there's also this gem that is part of a pipeline that computes the weighted
+average price for each day:
+
+```php
+$pipeline = [
+    [
+        '$addFields' => [
+            'weightedPrices' => [
+                '$map' => [
+                    'input' => '$prices',
+                    'as' => 'priceReport',
+                    'in' => [
+                        'duration' => [
+                            '$min' => [
+                                [
+                                    '$dateDiff' => [
+                                        'startDate' => '$$priceReport.previous.reportDate',
+                                        'endDate' => '$$priceReport.reportDate',
+                                        'unit' => 'second',
+                                    ],
+                                ],
+                                [
+                                    '$add' => [
+                                        [
+                                            '$multiply' => [
+                                                ['$hour' => '$$priceReport.reportDate'],
+                                                3600,
+                                            ],
+                                        ],
+                                        [
+                                            '$multiply' => [
+                                                ['$minute' => '$$priceReport.reportDate'],
+                                                60,
+                                            ],
+                                        ],
+                                        ['$second' => '$$priceReport.reportDate'],
+                                    ],
+                                ],
+                            ],
+                        ],
+                        'price' => '$$priceReport.previous.price',
+                    ],
+                ],
+            ],
+            'lastPrice' => ['$last' => '$prices'],
+        ],
+    ],
+];
+```
+
+Once again, the builder can make this a little more concise, but the complexity
+remains:
```

Comment:
```suggestion
The builder can make this a little more concise, but the complexity remains:
```

Many of the interstitial paragraphs use "once again" or similar language. This is so short that I don't think it'd needed.